src/Language/PureScript/Parser/Lexer.hs

```diff
@@ -173,6 +174,12 @@ updatePositions (x:xs) = x : zipWith update (x:xs) xs
 parseTokens :: Lexer u [PositionedToken]
 parseTokens = whitespace *> P.many parsePositionedToken <* P.skipMany parseComment <* P.eof
 
+lexLenient :: FilePath -> Text -> Either P.ParseError [PositionedToken]
+lexLenient f s = updatePositions <$> P.parse parseTokensLenient f s
+
+parseTokensLenient :: Lexer u [PositionedToken]
```

Comment:
I assume this is here to handle cases where something wouldn't fully parse normally? If so, could you also please add a test for that?